numpy/lib/tests/test_io.py

```diff
@@ -2799,3 +2800,20 @@ def test_load_multiple_arrays_until_eof():
     assert np.load(f) == 2
     with pytest.raises(EOFError):
         np.load(f)
+
+
+def test_savez_nopickle():
+    obj_array = np.array([1, 'hello'], dtype=object)
+    with temppath(suffix='.npz') as tmp:
+        np.savez(tmp, obj_array)
+
+    with temppath(suffix='.npz') as tmp:
```

Comment:
I couldn't find any warnings with Matti's suggestion, but since a few other places in the NumPy code base already use the old `tmpdir` `pytest` fixture, including `test_compressed_roundtrip` for `npz` files, perhaps worth a shot similar to below. It also has a nice context manager. One drawback is that `pytest` now prefers `tmp_path`, but that requires jumping through hoops for a context manger, but anyway `tmpdir` fixture is still actively used in NumPy suite. I don't now why they'd differ, but not sure how much we'd care if it works. Our in-house solution may not be as well tested on Windows as the `pytest` ones. Also, not only is the failing job Windows + Azure, but it is also PyPy on top of that. The `pytest` `7.4.0` used for that job is perhaps something that could be bumped as well, that's getting a bit older now.

```diff
--- a/numpy/lib/tests/test_io.py
+++ b/numpy/lib/tests/test_io.py
@@ -2802,14 +2802,14 @@ def test_load_multiple_arrays_until_eof():
         np.load(f)
 
 
-def test_savez_nopickle():
+def test_savez_nopickle(tmpdir):
     obj_array = np.array([1, 'hello'], dtype=object)
     with temppath(suffix='.npz') as tmp:
         np.savez(tmp, obj_array)
 
-    with temppath(suffix='.npz') as tmp:
+    with tmpdir.as_cwd():
         with pytest.raises(ValueError, match="Object arrays cannot be saved when.*"):
-            np.savez(tmp, obj_array, allow_pickle=False)
+            np.savez("tmp.npz", obj_array, allow_pickle=False)
 
     with temppath(suffix='.npz') as tmp:
         np.savez_compressed(tmp, obj_array)
```